### ISSUE TITLE
tests are broken

### ISSUE THREAD

**Issue by jugmac00** [zopefoundation/zc.zlibstorage#9]
I do not use this package - I just ran tox on all Zope projects:
https://jugmac00.github.io/blog/testing-the-tox-4-pre-release-at-scale/

```
❯ tox -e py36 -rvv
using tox.ini: /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/tox.ini (pid 17306)
using tox-3.23.0 from /home/jugmac00/opt/tox/lib/python3.8/site-packages/tox/__init__.py (pid 17306)
GLOB start: packaging 
GLOB sdist-make: /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/setup.py
[17314] /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage$ /home/jugmac00/opt/tox/bin/python setup.py sdist --formats=zip --dist-dir /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/dist >.tox/log/GLOB-0.log
running sdist
running egg_info
writing src/zc.zlibstorage.egg-info/PKG-INFO
writing dependency_links to src/zc.zlibstorage.egg-info/dependency_links.txt
writing entry points to src/zc.zlibstorage.egg-info/entry_points.txt
writing namespace_packages to src/zc.zlibstorage.egg-info/namespace_packages.txt
writing requirements to src/zc.zlibstorage.egg-info/requires.txt
writing top-level names to src/zc.zlibstorage.egg-info/top_level.txt
reading manifest file 'src/zc.zlibstorage.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/zc.zlibstorage.egg-info/SOURCES.txt'
running check
warning: check: missing required meta-data: url

creating zc.zlibstorage-1.2.0
creating zc.zlibstorage-1.2.0/src
creating zc.zlibstorage-1.2.0/src/zc
creating zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
creating zc.zlibstorage-1.2.0/src/zc/zlibstorage
copying files to zc.zlibstorage-1.2.0...
copying .travis.yml -> zc.zlibstorage-1.2.0
copying CHANGES.rst -> zc.zlibstorage-1.2.0
copying MANIFEST.in -> zc.zlibstorage-1.2.0
copying README.rst -> zc.zlibstorage-1.2.0
copying bootstrap.py -> zc.zlibstorage-1.2.0
copying buildout.cfg -> zc.zlibstorage-1.2.0
copying setup.py -> zc.zlibstorage-1.2.0
copying tox.ini -> zc.zlibstorage-1.2.0
copying src/zc/__init__.py -> zc.zlibstorage-1.2.0/src/zc
copying src/zc.zlibstorage.egg-info/PKG-INFO -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc.zlibstorage.egg-info/SOURCES.txt -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc.zlibstorage.egg-info/dependency_links.txt -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc.zlibstorage.egg-info/entry_points.txt -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc.zlibstorage.egg-info/namespace_packages.txt -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc.zlibstorage.egg-info/not-zip-safe -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc.zlibstorage.egg-info/requires.txt -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc.zlibstorage.egg-info/top_level.txt -> zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info
copying src/zc/zlibstorage/README.txt -> zc.zlibstorage-1.2.0/src/zc/zlibstorage
copying src/zc/zlibstorage/__init__.py -> zc.zlibstorage-1.2.0/src/zc/zlibstorage
copying src/zc/zlibstorage/component.xml -> zc.zlibstorage-1.2.0/src/zc/zlibstorage
copying src/zc/zlibstorage/tests.py -> zc.zlibstorage-1.2.0/src/zc/zlibstorage
Writing zc.zlibstorage-1.2.0/setup.cfg
creating /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/dist
creating '/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/dist/zc.zlibstorage-1.2.0.zip' and adding 'zc.zlibstorage-1.2.0' to it
adding 'zc.zlibstorage-1.2.0'
adding 'zc.zlibstorage-1.2.0/src'
adding 'zc.zlibstorage-1.2.0/CHANGES.rst'
adding 'zc.zlibstorage-1.2.0/bootstrap.py'
adding 'zc.zlibstorage-1.2.0/PKG-INFO'
adding 'zc.zlibstorage-1.2.0/tox.ini'
adding 'zc.zlibstorage-1.2.0/.travis.yml'
adding 'zc.zlibstorage-1.2.0/MANIFEST.in'
adding 'zc.zlibstorage-1.2.0/setup.py'
adding 'zc.zlibstorage-1.2.0/setup.cfg'
adding 'zc.zlibstorage-1.2.0/README.rst'
adding 'zc.zlibstorage-1.2.0/buildout.cfg'
adding 'zc.zlibstorage-1.2.0/src/zc'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info'
adding 'zc.zlibstorage-1.2.0/src/zc/zlibstorage'
adding 'zc.zlibstorage-1.2.0/src/zc/__init__.py'
adding 'zc.zlibstorage-1.2.0/src/zc/zlibstorage/tests.py'
adding 'zc.zlibstorage-1.2.0/src/zc/zlibstorage/__init__.py'
adding 'zc.zlibstorage-1.2.0/src/zc/zlibstorage/README.txt'
adding 'zc.zlibstorage-1.2.0/src/zc/zlibstorage/component.xml'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/namespace_packages.txt'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/SOURCES.txt'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/PKG-INFO'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/dependency_links.txt'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/entry_points.txt'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/requires.txt'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/not-zip-safe'
adding 'zc.zlibstorage-1.2.0/src/zc.zlibstorage.egg-info/top_level.txt'
removing 'zc.zlibstorage-1.2.0' (and everything under it)

GLOB finish: packaging  after 0.16 seconds
copying new sdistfile to '/home/jugmac00/.tox/distshare/zc.zlibstorage-1.2.0.zip'
package .tmp/package/1/zc.zlibstorage-1.2.0.zip links to dist/zc.zlibstorage-1.2.0.zip (/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox)
python3.6 (/usr/bin/python3.6) is {'executable': '/usr/bin/python3.6', 'implementation': 'CPython', 'version_info': [3, 6, 9, 'final', 0], 'version': '3.6.9 (default, Jan 26 2021, 15:33:00) \n[GCC 8.4.0]', 'is_64': True, 'sysplatform': 'linux', 'extra_version_info': None}
py36 uses /usr/bin/python3.6
py36 start: getenv /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36
py36 cannot reuse: -r flag
py36 create: /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36
setting PATH=/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin:/home/jugmac00/.cargo/bin:/home/jugmac00/.local/bin:/home/jugmac00/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[17317] /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox$ /home/jugmac00/opt/tox/bin/python -m virtualenv --no-download --python /usr/bin/python3.6 py36
created virtual environment CPython3.6.9.final.0-64 in 94ms
  creator CPython3Posix(dest=/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/jugmac00/.local/share/virtualenv)
    added seed packages: pip==21.0.1, setuptools==54.2.0, wheel==0.36.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
py36 installdeps: .[test], zope.testrunner
setting PATH=/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin:/home/jugmac00/.cargo/bin:/home/jugmac00/.local/bin:/home/jugmac00/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[17324] /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage$ /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin/python -m pip install '.[test]' zope.testrunner
Processing /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage
Collecting zope.testrunner
  Using cached zope.testrunner-5.3.0-py2.py3-none-any.whl (215 kB)
Requirement already satisfied: setuptools in ./.tox/py36/lib/python3.6/site-packages (from zc.zlibstorage==1.2.0) (54.2.0)
Collecting ZODB
  Using cached ZODB-5.6.0-py2.py3-none-any.whl (413 kB)
Collecting zope.interface
  Using cached zope.interface-5.4.0-cp36-cp36m-manylinux2010_x86_64.whl (251 kB)
Collecting zope.testing
  Using cached zope.testing-4.9-py2.py3-none-any.whl (50 kB)
Collecting manuel
  Using cached manuel-1.10.1-py3-none-any.whl
Collecting ZEO[test]
  Using cached ZEO-5.2.2-py2.py3-none-any.whl (288 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting zope.exceptions
  Using cached zope.exceptions-4.4-py2.py3-none-any.whl (18 kB)
Collecting zc.lockfile
  Using cached zc.lockfile-2.0-py2.py3-none-any.whl (9.7 kB)
Collecting persistent>=4.1.0
  Using cached persistent-4.7.0-cp36-cp36m-manylinux2010_x86_64.whl (249 kB)
Collecting transaction>=2.0.3
  Using cached transaction-3.0.1-py2.py3-none-any.whl (47 kB)
Collecting zdaemon
  Using cached zdaemon-4.3-py3-none-any.whl
Collecting ZConfig
  Using cached ZConfig-3.5.0-py2.py3-none-any.whl (134 kB)
Collecting msgpack<1
  Using cached msgpack-0.6.2-cp36-cp36m-manylinux1_x86_64.whl (249 kB)
Collecting random2
  Using cached random2-1.0.1-py3-none-any.whl
Collecting mock
  Using cached mock-4.0.3-py3-none-any.whl (28 kB)
Collecting cffi
  Using cached cffi-1.14.5-cp36-cp36m-manylinux1_x86_64.whl (401 kB)
Collecting BTrees>=4.2.0
  Using cached BTrees-4.8.0-cp36-cp36m-manylinux2010_x86_64.whl (3.5 MB)
Collecting zodbpickle>=1.0.1
  Using cached zodbpickle-2.0.0-cp36-cp36m-manylinux2010_x86_64.whl (331 kB)
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Building wheels for collected packages: zc.zlibstorage
  Building wheel for zc.zlibstorage (setup.py) ... done
  Created wheel for zc.zlibstorage: filename=zc.zlibstorage-1.2.0-py3-none-any.whl size=13375 sha256=3aba87aa5c080bd70d0515f7b5cb3cf12bd760e73edd014cc195c80a7a8618ae
  Stored in directory: /tmp/pip-ephem-wheel-cache-fxdcj596/wheels/91/75/56/936205b4ac6f75ac5f6105dff4e45ede7bc9fc1215d137022e
Successfully built zc.zlibstorage
Installing collected packages: pycparser, zope.interface, cffi, persistent, zodbpickle, ZConfig, zc.lockfile, transaction, six, BTrees, zope.exceptions, ZODB, zdaemon, zope.testrunner, zope.testing, ZEO, random2, msgpack, mock, manuel, zc.zlibstorage
Successfully installed BTrees-4.8.0 ZConfig-3.5.0 ZEO-5.2.2 ZODB-5.6.0 cffi-1.14.5 manuel-1.10.1 mock-4.0.3 msgpack-0.6.2 persistent-4.7.0 pycparser-2.20 random2-1.0.1 six-1.15.0 transaction-3.0.1 zc.lockfile-2.0 zc.zlibstorage-1.2.0 zdaemon-4.3 zodbpickle-2.0.0 zope.exceptions-4.4 zope.interface-5.4.0 zope.testing-4.9 zope.testrunner-5.3.0
py36 finish: getenv /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36 after 5.06 seconds
py36 start: installpkg /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/.tmp/package/1/zc.zlibstorage-1.2.0.zip
py36 inst: /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/.tmp/package/1/zc.zlibstorage-1.2.0.zip
write config to /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/.tox-config1 as '9b56b084b98cbe6dc10d740ad48815b9146a252cbfc3d78e9ffd42a36d965d1f /usr/bin/python3.6\n3.23.0 0 0 0\n00000000000000000000000000000000 .[test]\n00000000000000000000000000000000 zope.testrunner'
setting PATH=/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin:/home/jugmac00/.cargo/bin:/home/jugmac00/.local/bin:/home/jugmac00/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[17348] /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage$ /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin/python -m pip install --exists-action w .tox/.tmp/package/1/zc.zlibstorage-1.2.0.zip
Processing ./.tox/.tmp/package/1/zc.zlibstorage-1.2.0.zip
Requirement already satisfied: setuptools in ./.tox/py36/lib/python3.6/site-packages (from zc.zlibstorage==1.2.0) (54.2.0)
Requirement already satisfied: ZODB in ./.tox/py36/lib/python3.6/site-packages (from zc.zlibstorage==1.2.0) (5.6.0)
Requirement already satisfied: zope.interface in ./.tox/py36/lib/python3.6/site-packages (from zc.zlibstorage==1.2.0) (5.4.0)
Requirement already satisfied: BTrees>=4.2.0 in ./.tox/py36/lib/python3.6/site-packages (from ZODB->zc.zlibstorage==1.2.0) (4.8.0)
Requirement already satisfied: zc.lockfile in ./.tox/py36/lib/python3.6/site-packages (from ZODB->zc.zlibstorage==1.2.0) (2.0)
Requirement already satisfied: six in ./.tox/py36/lib/python3.6/site-packages (from ZODB->zc.zlibstorage==1.2.0) (1.15.0)
Requirement already satisfied: transaction>=2.4 in ./.tox/py36/lib/python3.6/site-packages (from ZODB->zc.zlibstorage==1.2.0) (3.0.1)
Requirement already satisfied: persistent>=4.4.0 in ./.tox/py36/lib/python3.6/site-packages (from ZODB->zc.zlibstorage==1.2.0) (4.7.0)
Requirement already satisfied: zodbpickle>=1.0.1 in ./.tox/py36/lib/python3.6/site-packages (from ZODB->zc.zlibstorage==1.2.0) (2.0.0)
Requirement already satisfied: ZConfig in ./.tox/py36/lib/python3.6/site-packages (from ZODB->zc.zlibstorage==1.2.0) (3.5.0)
Requirement already satisfied: cffi in ./.tox/py36/lib/python3.6/site-packages (from persistent>=4.4.0->ZODB->zc.zlibstorage==1.2.0) (1.14.5)
Requirement already satisfied: pycparser in ./.tox/py36/lib/python3.6/site-packages (from cffi->persistent>=4.4.0->ZODB->zc.zlibstorage==1.2.0) (2.20)
Building wheels for collected packages: zc.zlibstorage
  Building wheel for zc.zlibstorage (setup.py) ... done
  Created wheel for zc.zlibstorage: filename=zc.zlibstorage-1.2.0-py3-none-any.whl size=13375 sha256=7eb4b3db3b6b5ad49b883f03bd167ed5923275f087aaabdece68758bb6649aa9
  Stored in directory: /home/jugmac00/.cache/pip/wheels/be/6b/d6/f5d14b2497307462f73fa9a8748b5bec42ce8edb9db1f56d75
Successfully built zc.zlibstorage
Installing collected packages: zc.zlibstorage
  Attempting uninstall: zc.zlibstorage
    Found existing installation: zc.zlibstorage 1.2.0
    Uninstalling zc.zlibstorage-1.2.0:
      Successfully uninstalled zc.zlibstorage-1.2.0
Successfully installed zc.zlibstorage-1.2.0
py36 finish: installpkg /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/.tmp/package/1/zc.zlibstorage-1.2.0.zip after 1.06 seconds
py36 start: envreport 
setting PATH=/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin:/home/jugmac00/.cargo/bin:/home/jugmac00/.local/bin:/home/jugmac00/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[17361] /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage$ /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin/python -m pip freeze >.tox/py36/log/py36-0.log
py36 finish: envreport  after 0.22 seconds
py36 installed: BTrees==4.8.0,cffi==1.14.5,manuel==1.10.1,mock==4.0.3,msgpack==0.6.2,persistent==4.7.0,pycparser==2.20,random2==1.0.1,six==1.15.0,transaction==3.0.1,zc.lockfile==2.0,zc.zlibstorage @ file:///home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/.tmp/package/1/zc.zlibstorage-1.2.0.zip,ZConfig==3.5.0,zdaemon==4.3,ZEO==5.2.2,ZODB==5.6.0,zodbpickle==2.0.0,zope.exceptions==4.4,zope.interface==5.4.0,zope.testing==4.9,zope.testrunner==5.3.0
py36 start: run-test-pre 
py36 run-test-pre: PYTHONHASHSEED='253076749'
py36 finish: run-test-pre  after 0.00 seconds
py36 start: run-test 
py36 run-test: commands[0] | zope-testrunner --test-path=src -v1j99
setting PATH=/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin:/home/jugmac00/.cargo/bin:/home/jugmac00/.local/bin:/home/jugmac00/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[17363] /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage$ /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin/zope-testrunner --test-path=src -v1j99
Running tests at level 1
Running .EmptyLayer tests:
  Set up .EmptyLayer in 0.000 seconds.
  Running:

  Ran 0 tests with 0 failures, 0 errors and 0 skipped in 0.000 seconds.
Running zope.testrunner.layer.UnitTests tests:
  Running in a subprocess.
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZODB/blob.py:339: ResourceWarning: unclosed file <_io.FileIO name='/tmp/tmp4x_ve8_d/blobs/0x00/0x00/0x00/0x00/0x00/0x00/0x00/0x01/0x03df80ab63f4e288.blob' mode='rb' closefd=True>
  super(BlobFile, self).close()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZODB/blob.py:339: ResourceWarning: unclosed file <_io.FileIO name='/tmp/tmpt4wpl1_4/blobs/0x00/0x00/0x00/0x00/0x00/0x00/0x00/0x01/0x03df80ab6413eb88.blob' mode='rb' closefd=True>
  super(BlobFile, self).close()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZODB/blob.py:339: ResourceWarning: unclosed file <_io.FileIO name='/tmp/tmpma3h3n3w/blobs/0x00/0x00/0x00/0x00/0x00/0x00/0x00/0x01/0x03df80ab642c24aa.blob' mode='rb' closefd=True>
  super(BlobFile, self).close()
  Ran 49 tests with 0 failures, 0 errors and 0 skipped in 0.196 seconds.
  Tear down zope.testrunner.layer.UnitTests in 0.000 seconds.
Running .zlibstoragetests.FileStorageClientZlibZEOServerZlibTests tests:
  Running in a subprocess.
  Set up .zlibstoragetests.FileStorageClientZlibZEOServerZlibTests in 0.000 seconds.
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=19>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=13>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=20>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=21>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/StorageServer.py:604: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/FileStorageClientZlibZEOServerZlibTestsyiawryel/Data.fs'>
  self._iterators.pop(iid, None)


Failure in test checkTransactionExtensionFromIterator (zc.zlibstorage.tests.FileStorageClientZlibZEOServerZlibTests)
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File "/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZODB/tests/IteratorStorage.py", line 114, in checkTransactionExtensionFromIterator
    self.assertNotEqual(extension_bytes, txn.extension_bytes)
  File "/usr/lib/python3.6/unittest/case.py", line 838, in assertNotEqual
    raise self.failureException(msg)
AssertionError: b'\x80\x03}q\x00X\x03\x00\x00\x00fooq\x01K\x01s.' == b'\x80\x03}q\x00X\x03\x00\x00\x00fooq\x01K\x01s.'

  Ran 103 tests with 1 failures, 0 errors and 0 skipped in 25.686 seconds.
  Tear down .zlibstoragetests.FileStorageClientZlibZEOServerZlibTests in 0.000 seconds.
Running .zlibstoragetests.FileStorageClientZlibZEOZlibTests tests:
  Running in a subprocess.
  Set up .zlibstoragetests.FileStorageClientZlibZEOZlibTests in 0.000 seconds.
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=19>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=13>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=20>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=21>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/StorageServer.py:604: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/FileStorageClientZlibZEOZlibTestsiff44cyk/Data.fs'>
  self._iterators.pop(iid, None)


Failure in test checkTransactionExtensionFromIterator (zc.zlibstorage.tests.FileStorageClientZlibZEOZlibTests)
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File "/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZODB/tests/IteratorStorage.py", line 114, in checkTransactionExtensionFromIterator
    self.assertNotEqual(extension_bytes, txn.extension_bytes)
  File "/usr/lib/python3.6/unittest/case.py", line 838, in assertNotEqual
    raise self.failureException(msg)
AssertionError: b'\x80\x03}q\x00X\x03\x00\x00\x00fooq\x01K\x01s.' == b'\x80\x03}q\x00X\x03\x00\x00\x00fooq\x01K\x01s.'

  Ran 103 tests with 1 failures, 0 errors and 0 skipped in 25.935 seconds.
  Tear down .zlibstoragetests.FileStorageClientZlibZEOZlibTests in 0.000 seconds.
Running .zlibstoragetests.FileStorageZEOZlibTests tests:
  Running in a subprocess.
  Set up .zlibstoragetests.FileStorageZEOZlibTests in 0.000 seconds.
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=19>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=13>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=20>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/_forker.py:181: ResourceWarning: unclosed file <_io.BufferedRandom name=21>
  gc.collect()
/home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/lib/python3.6/site-packages/ZEO/StorageServer.py:604: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/FileStorageZEOZlibTestsmankqby_/Data.fs'>
  self._iterators.pop(iid, None)
  Ran 103 tests with 0 failures, 0 errors and 0 skipped in 25.811 seconds.
  Tear down .zlibstoragetests.FileStorageZEOZlibTests in 0.000 seconds.
Running .zlibstoragetests.FileStorageZlibRecoveryTest tests:
  Running in a subprocess.
  Set up .zlibstoragetests.FileStorageZlibRecoveryTest in 0.000 seconds.
  Ran 4 tests with 0 failures, 0 errors and 0 skipped in 0.240 seconds.
  Tear down .zlibstoragetests.FileStorageZlibRecoveryTest in 0.000 seconds.
Running .zlibstoragetests.FileStorageZlibTests tests:
  Running in a subprocess.
  Set up .zlibstoragetests.FileStorageZlibTests in 0.000 seconds.
  Ran 100 tests with 0 failures, 0 errors and 0 skipped in 17.950 seconds.
  Tear down .zlibstoragetests.FileStorageZlibTests in 0.000 seconds.
Running .zlibstoragetests.FileStorageZlibTestsWithBlobsEnabled tests:
  Running in a subprocess.
  Set up .zlibstoragetests.FileStorageZlibTestsWithBlobsEnabled in 0.000 seconds.
  Ran 100 tests with 0 failures, 0 errors and 0 skipped in 18.565 seconds.
  Tear down .zlibstoragetests.FileStorageZlibTestsWithBlobsEnabled in 0.000 seconds.
Running zc.zlibstorage.tests.ZLibHackLayer tests:
  Running in a subprocess.
  Set up zc.zlibstorage.tests.ZLibHackLayer in 0.000 seconds.
  Ran 100 tests with 0 failures, 0 errors and 0 skipped in 18.504 seconds.
  Tear down zc.zlibstorage.tests.ZLibHackLayer in 0.000 seconds.
Tearing down left over layers:
  Tear down .EmptyLayer in 0.000 seconds.

Tests with failures:
   checkTransactionExtensionFromIterator (zc.zlibstorage.tests.FileStorageClientZlibZEOServerZlibTests)
   checkTransactionExtensionFromIterator (zc.zlibstorage.tests.FileStorageClientZlibZEOZlibTests)
Total: 662 tests, 2 failures, 0 errors and 0 skipped in 26.487 seconds.
ERROR: InvocationError for command /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/py36/bin/zope-testrunner --test-path=src -v1j99 (exited with code 1)
py36 finish: run-test  after 26.78 seconds
py36 start: run-test-post 
py36 finish: run-test-post  after 0.00 seconds
__________________________________________ summary ___________________________________________
ERROR:   py36: commands failed
cleanup /home/jugmac00/All/output_zope/zopefoundation/zc.zlibstorage/.tox/.tmp/package/1/zc.zlibstorage-1.2.0.zip
```

**Comment by navytux** [comment]
Partly fixed by: https://github.com/zopefoundation/zc.zlibstorage/pull/10. See also: https://github.com/zopefoundation/zc.zlibstorage/pull/12. /cc @icemac

**Comment by navytux** [comment]
Tests should be fixed by https://github.com/zopefoundation/zc.zlibstorage/pull/13.

**Comment by icemac** [comment]
Fixed by #13. Thank you!
